integrations/opencost/integration.py

```diff
@@ -67,6 +66,18 @@ def validate(cls, value: str) -> None:
             )
 
 
+class CloudCostAggregateField(str):
+    @classmethod
+    def validate(cls, value: str) -> None:
+        # Regular expression to validate the format of the aggregation value
+        regex = r"^((invoiceEntityID|accountID|provider|providerID|category|service)(,(invoiceEntityID|accountID|provider|providerID|category|service))*)*$"
+        if not re.match(regex, value):
+            raise ValueError(
+                "Invalid aggregation format. Use 'invoiceEntityID', 'accountID', 'provider', "
+                "'providerID', 'category', 'service' or comma-separated list of values."
+            )
+
+
```

Comment:
https://stackoverflow.com/a/76330547
lets use this instead with literal for the list typing

Reply:
Thanks for the suggestion. The usecases are different though. In this case, they are still treated as a string as the comma separated literal must still remain as a string to be passed to the OpenCost API. Am I missing something?

Reply:
then make sure the desirialize is reverting the format back